File: Assets/Scripts/App/Data/DataClass/Map/SpawnMonsterData.cs
using App.Static.Enum;
using XLua;

namespace App.Data.DataClass.Map
{
    [LuaCallCSharp]
    public class SpawnMonsterData
    {
        public float hard;
        public EMonsterType monsterType;
        public float nextSpawnTime;
    }
}

File: Assets/Scripts/App/Lua/App/Generic/SpawnMonsterData.lua
SpawnMonsterData = Object:SubClass('SpawnMonsterData')

SpawnMonsterData.monsterType = nil
SpawnMonsterData.nextSpawnTime = 0
SpawnMonsterData.hard = 0

function SpawnMonsterData:Construct(args)
    self.monsterType = args[1]
    self.nextSpawnTime = args[2]
    self.hard = args[3]
end